pkg/instance/instance.go

```diff
@@ -59,3 +62,29 @@ func (i *Info) UpgradeRequired() bool {
 	// fully deconfigured before being configured by the current version.
 	return true
 }
+
+// GetIPv4Address returns an IPv4 address of the instance. An error will be thrown if the instance object has
+// an IPv6 address or a DNS address that does not resolve to an IPv4 network address.
+func (i *Info) GetIPv4Address() (string, error) {
+	if ip := net.ParseIP(i.Address); ip != nil {
+		// Address is either an IPv6 or IPv4 address
+		ipv4 := ip.To4()
```

Comment:
you can directly do:
if net.ParseIP(i.Address).To4()  != nil {
}

Reply:
if `net.ParseIP(i.Address)` returns `nil` wouldn't calling `.To4()` on it cause a panic?

Reply:
It does not, tried it [here](https://play.golang.org/p/-uCW6RvUGQg)

Reply:
However, as per the [docs](https://pkg.go.dev/net#ParseIP):
>If s is not a valid textual representation of an IP address, ParseIP returns nil.

Reply:
I will opt to keep this as is; allows for more accurate error reporting.

Reply:
More than accurate error reporting, `net.Parse()` can return null if the string is not a valid IP address.